tests/51media/03ascii.pl

```diff
@@ -2,80 +2,58 @@
 my $content_uri;
 
 test "Can upload with ASCII file name",
-   requires => [ $main::API_CLIENTS[0], local_user_fixture() ],
+   requires => [ local_user_fixture() ],
 
    do => sub {
-      my ( $http, $user ) = @_;
-
-      $http->do_request(
-         method       => "POST",
-         full_uri     => "/_matrix/media/r0/upload",
-         content      => "Test media file",
-         content_type => "text/plain",
-
-         params => {
-            access_token => $user->access_token,
-            filename => "ascii",
-         }
-      )->then( sub {
-         my ( $body ) = @_;
+      my ( $user ) = @_;
+      upload_test_content( $user, filename=>"ascii" )->then( sub {
+         ( $content_id, $content_uri ) = @_;
+         Future->done(1);
+      });
+   };
 
-         assert_json_keys( $body, qw( content_uri ));
+# we only need one user for these tests
+my $user_fixture = local_user_fixture();
 
-         $content_uri = $body->{content_uri};
+my %TEST_CASES = (
+   "ascii" => "inline; filename=ascii",
 
-         my $parsed_uri = URI->new( $body->{content_uri} );
-         my $server = $parsed_uri->authority;
-         my $path = $parsed_uri->path;
+   # synapse uses filename* encoding for this, though regular filename encoding
+   # with a quoted string would be valid too
+   "name with spaces" => "inline; filename*=utf-8''name%20with%20spaces",
```

Comment:
It would be nice if this was a map of file names to a list of valid content dispositions, so we don't force synapse's choice on other projects

Reply:
that's not quite as easy as it sounds. have made it parse the header.